fsharp-backend/src/BackendOnlyStdLib/HttpBytesClient.fs

```diff
@@ -0,0 +1,212 @@
+    UseProxy = true,
+    Proxy = System.Net.WebProxy(Config.httpclientProxyUrl, false),
+
+    // Don't add a RequestId header for opentelemetry
+    ActivityHeadersPropagator = null,
+
+    // Users share the HttpClient, don't let them share cookies!
+    UseCookies = false
+  )
+
+let private httpClient : HttpClient =
+  new HttpClient(
+    socketHandler,
+    disposeHandler = false,
+
+    // These seem like reasonable defaults
+    Timeout = System.TimeSpan.FromSeconds 30.0,
+    MaxResponseContentBufferSize = 1024L * 1024L * 100L // 100MB
+  )
+
+let private httpCall
+  (url : string)
+  (queryParams : (string * string list) list)
+  (method : HttpMethod)
+  (reqHeaders : HttpHeaders.T)
+  (reqBody : byte array)
+  : Task<Result<HttpResult, ClientError>> =
+  task {
+    use _ =
+      Telemetry.child
+        "HttpClient.call"
+        [ "request.url", url; "request.method", method ]
+    try
+      let uri = System.Uri(url, System.UriKind.Absolute)
+
+      // currently we only support http(s) requests
+      if uri.Scheme <> "https" && uri.Scheme <> "http" then
+        return Error { code = None; error = "Unsupported protocol" }
+      else
+        let reqUri =
+          let queryString =
+            let query =
+              // Remove the question mark
+              if uri.Query.Length > 0 then String.dropLeft 1 uri.Query else uri.Query
+            HttpQueryEncoding.createQueryString query queryParams
+
+          System.UriBuilder(
+            Scheme = uri.Scheme,
+            Host = uri.Host,
+            Port = uri.Port,
+            Path = uri.AbsolutePath,
+            Query = queryString
+          )
+
+        use req =
+          new HttpRequestMessage(
+            method,
+            string reqUri,
+            Version = System.Net.HttpVersion.Version30,
+            Content = new ByteArrayContent(reqBody)
+          )
+
+        // headers
+        // TODO is it still appropriate to include these default headers?
+        let defaultHeaders =
+          Map [ "Accept", "*/*"; "Accept-Encoding", "deflate, gzip, br" ]
+
+        Map reqHeaders
+        |> Map.mergeFavoringRight defaultHeaders
+        |> Map.iter (fun k v ->
+          // TODO do we need this branch?
+          // I think so - .NET is odd with content types, to my memory
+          if String.equalsCaseInsensitive k "content-type" then
+            try
+              req.Content.Headers.ContentType <-
+                Headers.MediaTypeHeaderValue.Parse(v)
+            with
+            | :? System.FormatException ->
+              Exception.raiseCode "Invalid content-type header"
+          else
+            // Dark headers can only be added once, as they use a Dict.
+            // Remove them so they don't get added twice.
+            // TODO: re-evaluate if we need this. See logic in HttpClient.
+            // Also, consider that we could (do, in this draft) return headers
+            // as a list of tuples, so the Dict claim is false currently.
```

Comment:
I agree, this shouldn't use a dict, and we should allow headers to be set multiple times